frontend/src/pages/modelRegistry/ModelRegistryCoreLoader.tsx

```diff
@@ -31,9 +31,15 @@ const ModelRegistryCoreLoader: React.FC<ModelRegistryCoreLoaderProps> =
       modelRegistriesLoaded,
       modelRegistriesLoadError,
       modelRegistries,
-      preferredModelRegistry,
+      updatePreferredModelRegistry,
     } = React.useContext(ModelRegistrySelectorContext);
 
+    const preferredModelRegistry = modelRegistries.find((mr) => mr.metadata.name === modelRegistry);
+
+    React.useEffect(() => {
+      updatePreferredModelRegistry(preferredModelRegistry);
+    }, [preferredModelRegistry, updatePreferredModelRegistry]);
```

Comment:
yeah, I tried to address this issue in different way. But when this logic doesn't seems to work, when we reload the page and it will still point to the `firstModelRegistry`. I have updated this logic a bit. Let me know what you think.